code_files/weather_reporter.py

```diff
@@ -0,0 +1,62 @@
+from computations import Computations
+
+MONTHS = {
+    1: "January",
+    2: "February",
+    3: "March",
+    4: "April",
+    5: "May",
+    6: "June",
+    7: "July",
+    8: "August",
+    9: "September",
+    10: "October",
+    11: "November",
+    12: "December",
+}
+
+class WeatherReporter:
+    @staticmethod
+    def generate_report_extremes(readings):
+        """Generates a report summarizing weather extremes for a year."""
+
+        h_temp, l_temp, h_humidity = Computations.compute_extreme_stats(readings)
+        date_htemp = h_temp.date.split("-")
+        date_ltemp = l_temp.date.split("-")
+        date_hhumid = h_humidity.date.split("-")
+
+        print(f"\n<==== Report for the year {date_htemp[0]}: ====>")
+        print(f"Highest: {h_temp.max_temp}C on {MONTHS[int(date_htemp[1])]} {date_htemp[2]}")
+        print(f"Lowest: {l_temp.min_temp}C on {MONTHS[int(date_ltemp[1])]} {date_ltemp[2]}")
+        print(f"Humidity: {h_humidity.max_humidity}% on {MONTHS[int(date_hhumid[1])]} {date_hhumid[2]}")
+        print("-------------------------------------")
```

Comment:
Let's only use `WeatherReporter` for printing only. All calculations should be moved to calculations class.

Reply:
[Latest commit](https://github.com/ttqureshi/WeatherMan/commit/a9bfba76030779f62875a46887178ea2565ca521)